src/colab_superarchives/views.py

```diff
@@ -418,43 +419,86 @@ def get_object(self, *args, **kwargs):
+
+        return super(ManageUserSubscriptionsView,
+                     self).get_context_data(**context)
+
     def post(self, request, *args, **kwargs):
         user = self.get_object()
-        for email in user.emails.values_list('address', flat=True):
-            lists = self.request.POST.getlist(email)
-            info_messages = mailman.update_subscription(email, lists)
-            for msg_type, message in info_messages:
-                show_message = getattr(messages, msg_type)
-                show_message(request, _(message))
 
-        return redirect('user_profile', username=user.username)
+        # If it is an ajax request, get the user mailling lists
+        # to display filtering it by name
+        if request.is_ajax():
+            response = self.search_lists_by_name(user,
+                                                 request.POST.get('listname'))
+        else:
+            for email in user.emails.values_list('address', flat=True):
+                lists = self.request.POST.getlist(email)
+                info_messages = mailman.update_subscription(email, lists)
+                for msg_type, message in info_messages:
+                    show_message = getattr(messages, msg_type)
+                    show_message(request, _(message))
 
-    def get_context_data(self, **kwargs):
+            response = redirect('user_profile', username=user.username)
+
+        return response
+
+    def search_lists_by_name(self, user, listname):
+        context = self.get_mailling_lists(user, listname)
+
+        html = render_to_string(u'accounts/subscription_lists.html',
+                                {'membership': context['membership']})
+
+        return http.HttpResponse(html)
+
+    def get_mailling_lists(self, user, listname=None):
         context = {}
         context['membership'] = {}
 
-        user = self.get_object()
         emails = user.emails.values_list('address', flat=True)
-        all_lists = mailman.all_lists()
 
         for email in emails:
-            lists = []
-            lists_for_address = mailman.mailing_lists(address=email,
-                                                      names_only=True)
-            for mlist in all_lists:
-                if mlist.get('listname') in lists_for_address:
-                    checked = True
-                else:
-                    checked = False
+
+            lists = self.filter_lists_by_name(email, listname)
+            context['membership'].update({email: lists})
+
+        return context
+
+    def filter_lists_by_name(self, user_email, listname):
+        all_lists = mailman.all_lists()
+        lists = []
+        lists_for_address = mailman.mailing_lists(address=user_email,
+                                                  names_only=True)
+        for mlist in all_lists:
+
+            checked = self.check_user_list(mlist, lists_for_address)
+
+            if listname is not None and listname in mlist.get('listname'):
```

Comment:
@italopaiva, @Emiliemorais I simplified this if-else to this one here: https://github.com/colab/colab-superarchives-plugin/blob/master/src/colab_superarchives/views.py#L488
hope you guys don't mind : )